Sources/OpenAPIRuntime/Interface/Body.swift

```diff
@@ -0,0 +1,535 @@
+    /// - Parameter transform: A mapping closure.
+    /// - Throws: If a known length was provided to this body at
+    /// creation time, the transform closure must not change the length of
+    /// each chunk.
+    public func mapChunks(
+        _ transform: @escaping @Sendable (Element) async -> Element
+    ) -> Body {
+        let validatedTransform: @Sendable (Element) async -> Element
+        switch length {
+        case .known:
+            validatedTransform = { element in
+                let transformedElement = await transform(element)
+                guard transformedElement.count == element.count else {
+                    fatalError(
+                        "OpenAPIRuntime.Body.mapChunks transform closure attempted to change the length of a chunk in a body which has a total length specified, this is not allowed."
+                    )
+                }
+                return transformedElement
+            }
+        case .unknown:
+            validatedTransform = transform
+        }
+        return Body(
+            sequence: map(validatedTransform),
+            length: length,
+            iterationBehavior: iterationBehavior
+        )
+    }
+}
+
+// MARK: - Consumption utils
+
+extension Body {
+
+    /// An error thrown by the `collect` function when the body contains more
+    /// than the maximum allowed number of bytes.
+    private struct TooManyBytesError: Error, CustomStringConvertible, LocalizedError {
+        let maxBytes: Int
+
+        var description: String {
+            "OpenAPIRuntime.Body contains more than the maximum allowed \(maxBytes) bytes."
+        }
+
+        var errorDescription: String? {
+            description
+        }
+    }
+
+    /// An error thrown by the `collect` function when another iteration of
+    /// the body is not allowed.
+    private struct TooManyIterationsError: Error, CustomStringConvertible, LocalizedError {
+
+        var description: String {
+            "OpenAPIRuntime.Body attempted to create a second iterator, but the underlying sequence is only safe to be iterated once."
+        }
+
+        var errorDescription: String? {
+            description
+        }
+    }
+
+    /// Accumulates the full body in-memory into a single buffer
+    /// up to `maxBytes` and returns it.
+    /// - Parameters:
+    ///   - maxBytes: The maximum number of bytes this method is allowed
+    ///     to accumulate in memory before it throws an error.
+    /// - Throws: `TooManyBytesError` if the the sequence contains more
+    ///   than `maxBytes`.
+    public func collect(upTo maxBytes: Int) async throws -> DataType {
+
+        // As a courtesy, check if another iteration is allowed, and throw
+        // an error instead of fatalError here if the user is trying to
+        // iterate a sequence for the second time, if it's only safe to be
+        // iterated once.
+        if iterationBehavior == .single {
```

Comment:
Same here. Shouldn't have to do that.

Reply:
Fair, again here, it's meant to provide a better, more descriptive error that helps the adopter know what's wrong and how to fix it. Also, it throws an error here instead of crashing (as `makeAsyncIterator` isn't throwing, so we can't throw there), which again means that when the unlikely case happens (retry, redirect), the adopter's server/app isn't taken down, but they just get an error in their logs.